doc/api/stream.md

```diff
@@ -1843,10 +1843,11 @@ or write buffered data before a stream ends.
 It is recommended that errors occurring during the processing of the
 `writable._write()` and `writable._writev()` methods are reported by invoking
 the callback and passing the error as the first argument. This will cause an
-`'error'` event to be emitted by the `Writable`. Throwing an `Error` from within
-`writable._write()` can result in unexpected and inconsistent behavior depending
-on how the stream is being used. Using the callback ensures consistent and
-predictable handling of errors.
+`'error'` event to be emitted by the `Writable`, unless the `'error'` event has
+already been previously emitted on the instance. Throwing an `Error` from
+within `writable._write()` can result in unexpected and inconsistent behavior
+depending on how the stream is being used. Using the callback ensures
+consistent and predictable handling of errors.
```

Comment:
In the future... I'm wondering if we shouldn't explore wrapping the call with a try/catch that will automatically catch and forward errors to the callback. Obviously that would be semver-major. What do you think @mcollina

Reply:
I suspect the reason we don't already do this is that in the past try/catch was slow in V8. Which is not necessarily true anymore. I think it's a great idea.

We could do something similar with `read`, i.e. allow the user to throw normally and catch it.